staging/src/k8s.io/apiextensions-apiserver/examples/client-go/pkg/client/clientset/versioned/clientset.go

```diff
@@ -51,9 +53,17 @@ func (c *Clientset) Discovery() discovery.DiscoveryInterface {
 }
 
 // NewForConfig creates a new Clientset for the given config.
+// If config's RateLimiter is not set and QPS and Burst are acceptable,
+// NewForConfig will generate a shared rate-limiter for config.
 func NewForConfig(c *rest.Config) (*Clientset, error) {
 	configShallowCopy := *c
-	if configShallowCopy.RateLimiter == nil && configShallowCopy.QPS > 0 {
+	if configShallowCopy.RateLimiter == nil && configShallowCopy.QPS < 0 {
```

Comment:
we cannot tighten this validation. -1 means "do not rate limit". see test for this at https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-base/config/validation/validation_test.go#L50-L53

Reply:
I will fix it.